src/test/scala/org/broadinstitute/hail/TestUtils.scala

```diff
@@ -41,5 +41,16 @@ object TestUtils {
 
     new VariantDataset(VariantMetadata(sampleIds), rdd)
   }
+
+  def assertVectorEqualityDouble(A: Vector[Double], B: Vector[Double], tolerance: Double = 1e-6) {
+    assert(A.size == B.size)
+    assert((0 until A.size).forall(i => D_==(A(i), B(i), tolerance)))
+  }
+
+  def assertMatrixEqualityDouble(A: Matrix[Double], B: Matrix[Double], tolerance: Double = 1e-6) {
+    assert(A.rows == B.rows)
+    assert(A.cols == B.cols)
+    assert((0 until A.rows).forall(i => (0 until A.cols).forall(j => D_==(A(i, j), B(i, j), tolerance))))
```

Comment:
I'm uncomfortable having a default tolerance in two places. However, Scala doesn't provide a great way to abstract over this.

I think the easiest way to deal with this is to have a `defaultTolerance` value in [utils/package.scala](https://github.com/hail-is/hail/blob/master/src/main/scala/org/broadinstitute/hail/utils/package.scala) which is used as the default value for all the `D_==` type methods and is used as the default tolerance here.